zeebe/logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/serializer/DataFrameDescriptor.java

```diff
@@ -23,9 +23,11 @@ public final class DataFrameDescriptor {
 
   public static final int HEADER_LENGTH = 12;
 
-  public static void setFramedLength(
-      final MutableDirectBuffer buffer, final int offset, final int length) {
+  public static void write(final MutableDirectBuffer buffer, final int offset, final int length) {
     buffer.putInt(offset + FRAME_LENGTH_OFFSET, length);
+    for (int i = 4; i < HEADER_LENGTH; i++) {
```

Comment:
💭 no need to do a ~ `memcopy` here probably as the compiler should just unroll this loop, there is no data dependencies between iteration and the number of iterations is static.